examples/indexer/java/LookupApplication.java

```diff
@@ -6,43 +6,150 @@
 import com.algorand.algosdk.v2.client.common.Client;
 import com.algorand.algosdk.crypto.Address;
 import org.json.JSONObject;
+import com.algorand.algosdk.v2.client.model.ApplicationResponse;
+import com.algorand.algosdk.v2.client.common.Response;
 
 public class LookupApplication {
     public Client indexerInstance = null;
     // utility function to connect to a node
     private Client connectToNetwork(){
         final String INDEXER_API_ADDR = "localhost";
-        final int INDEXER_API_PORT = 59998;       
-        IndexerClient indexerClient = new IndexerClient(INDEXER_API_ADDR, INDEXER_API_PORT); 
+        final int INDEXER_API_PORT = 8981;
+        IndexerClient indexerClient = new IndexerClient(INDEXER_API_ADDR, INDEXER_API_PORT);
         return indexerClient;
     }
+
     public static void main(String args[]) throws Exception {
         LookupApplication ex = new LookupApplication();
         IndexerClient indexerClientInstance = (IndexerClient)ex.connectToNetwork();
-        Long application_id = Long.valueOf(22);
-        String response = indexerClientInstance.lookupApplicationByID(application_id).execute().toString();
-        JSONObject jsonObj = new JSONObject(response.toString());
-        System.out.println("Response Info: " + jsonObj.toString(2)); // pretty print json
+        Long application_id = Long.valueOf(15974179);   
+        Response<ApplicationResponse> response = indexerClientInstance
+            .lookupApplicationByID(application_id).execute(); 
+        if (!response.isSuccessful()) {
+            throw new Exception(response.message());
+        } 
+        JSONObject jsonObj = new JSONObject(response.body().toString());
+        System.out.println("Response Info: " + jsonObj.toString(2)); // pretty print json          
+
     }
  }
-
-// response information should look similar to this...
-//  Response Info:
-//  {
-//   "application": {
-//     "id": 22,
-//     "params": {
-//       "global-state": [],
-//       "creator": "GHFRLVOMKJNTJ4HY3P74ZR4CNE2PB7CYAUAJ6HVAVVDX7ZKEMLJX6AAF4M",
-//       "local-state-schema": {
-//         "num-uint": 0,
-//         "num-byte-slice": 0
-//       },
-//       "global-state-schema": {
-//         "num-uint": 0,
-//         "num-byte-slice": 0
+// response should look similar to this...
```

Comment:
This response is surely better than what we had. However, it is a bit lengthy and without context. Please consider using the "Hello World" application, as it will be much more concise and you may reference the this page for additional context for the keys and programs:
https://developer.algorand.org/docs/features/asc1/stateful/sdks/

Reply:
as for the long response, ya, sometimes after a few repeating groups in the response, i just chop it off and put a "..." after it, also feel free to add this link in the comment. for more info on applications, see. this link https://developer.algorand.org/docs/features/asc1/stateful/sdks/